### ISSUE TITLE
Typings not being picked up after installing 0.10.10

### ISSUE THREAD

**Issue by gregvanl** [Microsoft/vscode#3936]
I found this going through the /docs/runtimes/Node.js walkthrough after seeing a doc feedback verbatim saying that typings weren't being picked up.
After creating an express application, trying to get IntelliSense for node and express types fails.
If I roll-back to 0.10.8, the typings are picked up correctly.
The generated express app does not include a jsconfig.json file.

**Comment by gregvanl** [comment]
This may be a simple repro for https://github.com/Microsoft/vscode/issues/3791

**Comment by gregvanl** [comment]
I got it working again by add a jsconfig.json to the root with just the module set:
{
    "compilerOptions": {
        "module": "commonjs"
    }
}
We'll need to update the Node.js walkthrough if this becomes a necessary step. https://github.com/Microsoft/vscode-docs/issues/221

**Comment by tstringer** [comment]
:+1: I just ran into this same problem and adding `jsconfig.json` *and restarting VS Code* indeed fixed IntelliSense. Is this by design?

**Comment by egamma** [comment]
This is the underlying issue Microsoft/TypeScript#292, that is, the default of the module type was changed by accident, this will be fixed for TS 1.9. For now the work around is to add a jsconfig.json.

**Comment by tonyfi** [comment]
I run into the same problem. After adding jsconfig.json IntelliSense is working. However warnings are not displayed. For example, if we pass a string parameter instead of number as shown in the Node.js walkthrough there is no warning.

**Comment by egamma** [comment]
@tonyfi We have decided to deprecate the built-in linting support in favor of existing linters like `eslint` or `jshint`. These linters are more powerful, extensible, and many of our users are already using them. We understand that if you are not using a linter already then the price for this is that you have to install a linter and configure it.

I've tried to document this in the [release notes](http://code.visualstudio.com/updates/#_languages-javascript) see the `migration section`.

**Comment by egamma** [comment]
@tstringer actually this is by design. When you have no jsconfig.json or tsconfig.json, then each file is treated as an island. The file can have references to other files e.g. using commonjs imports and they will be considered. However, other files in the same folder like the typings folder are not considered since there is no reference to them. You can add a reference to a typings file manually using a /// reference `/// <reference path="typings/node/node.d.ts" />`.

VS Code 0.10.8 used a slightly different semantics. When there was no jsconfig.json then all files inside the workspace where considered to belong to the same project. Now that TypeScript supports hybrid projects of JS and TS files, it is no longer an option to have different project semantics and you now always get the TypeScript semantic.

Bottom line you should always add a jsconfig.json to define the project context and its excluded files. This will give you reference searches, Intellisense etc.

For March we have added some hinting UI to guide the user to create a jsconfig.json.

**Comment by tstringer** [comment]
@egamma Thanks so much for the information!  Just to clarify, is this no longer considered a bug or accidental-break?  And if that's the case, for JavaScript projects/applications going forward, the correct approach would be to create the `jsconfig.json` file in the project root 100% of the time?

Likewise, if this is the intended behavior, is this documented?  I couldn't find any documentation explaining this prior to my initial comment.

Thanks so much in advance for your time!!  When I get my head wrapped around the intention and future of this, I'll make an addendum to a [blog post](http://tstringer.github.io/vscode/2016/03/10/vscode-intellisense-breaks-0.10.10.html) I released about this.

**Comment by egamma** [comment]
@tstringer correct, this is as designed. Salsa doesn't automatically pick up .d.ts files.

In almost all the cases a `jsconfig.json` is the right thing to do.  We need to document the difference between a js project without a jsconfig.json and one that has a jsconfig.json. I've create https://github.com/Microsoft/vscode-docs/issues/233 to track this.

**Comment by tstringer** [comment]
@egamma Perfect! Thanks! I'll go ahead and update my blog post so it doesn't mislead readers into thinking it is "broken" or a "bug".
